src/components/DeckMechanisms/handleCardEffect.js

```diff
@@ -292,41 +302,81 @@ const handleCardEffect = (state, card, mode) => {
       break
     }
 
+    // Harvester of Souls
+    case 'N38': {
+      const copiedCard = getHarvesterOfSoulsCopiedCard(state.deck)
+      state.deck.push(copiedCard)
+      break
+    }
+
     default:
   }
 
   return state
 }
 
-function getNonPirates(state) {
-  return cardId => {
-    return state.deck.find(card => card.id === cardId).race !== 'pirate'
-  }
+function getNonPiratesInHand(hand, deck) {
+  return hand.filter(
+    cardInHand =>
+      deck.find(deckCard => areCardsEqual(cardInHand, deckCard)).race !==
+      'pirate'
+  )
 }
 
-function getSatyrs(state) {
-  return card => !state.hand.includes(card.id) && card.race === 'satyr'
+function getSatyrs(hand, deck) {
+  return deck.filter(
+    card =>
+      !hand.find(cardInHand => areCardsEqual(cardInHand, card)) &&
+      card.race === 'satyr'
+  )
 }
 
-function getPlayableSpells(state) {
-  return cardId => {
-    if (cardId === 'W1') {
-      return state.specifics.frozenEnemiesLevel !== 0
+function getPlayableSpells(hand, deck, frozenEnemiesLevel) {
+  return hand.filter(cardInHand => {
+    if (cardInHand.id === 'W1') {
+      return frozenEnemiesLevel !== 0
     }
 
-    const cardInDeck = state.deck.find(card => card.id === cardId)
-
+    const cardInDeck = deck.find(deckCard =>
+      areCardsEqual(cardInHand, deckCard)
+    )
     return cardInDeck.type === 'spell'
-  }
+  })
 }
 
-function getCollectorMirzToken(level) {
+function getCollectorMirzToken(deck, level) {
   const id = 'T' + arrayRandom([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 13, 14])
   const token = resolveCardForLevel({ id })
   token.level = [5, 6, 6, 8, 10][level - 1]
   token.weight = 0
-  token.id = id + ':' + Math.random().toString(36).substring(7)
+  token.id = id
+  token.idx = deck.filter(card => card.id === id).length.toString()
+  token.created = true
   return token
 }
 
+function getHarvesterOfSoulsCopiedCard(deck) {
+  const id = arrayRandom(
+    cards
+      .filter(card => card.type === 'unit' && card.id !== 'T12')
```

Comment:
```suggestion
      .filter(card => card.type === 'unit')
```

Qordia’s nests are no longer units.